### PR TITLE
feat(tool): add GitHub integration

### LINKED ISSUE #4352: feat: Add GitHub integration tool
## Summary

Implement GitHub integration for ZeroClaw agent to enable interaction with GitHub repositories, issues, pull requests, and GitHub Actions workflows.

## Motivation

ZeroClaw currently lacks native GitHub integration. Users need to manually configure external tools or use browser automation to interact with GitHub. A native integration would provide:

- Direct access to repository information
- Issue management (create, read, comment)
- Pull request operations (list, read, merge)
- GitHub Actions workflow management

## Proposed Solution

Add a `github` tool that implements the following actions:

### Read Operations (enabled by default)
- `get_repo` - Get repository information
- `list_repos` - List user's or organization's repositories  
- `list_issues` - Search issues with filters
- `get_issue` - Get single issue details
- `list_prs` - List pull requests
- `get_pr` - Get pull request details
- `list_workflows` - List GitHub Actions workflows
- `list_runs` - List workflow runs

### Write Operations (require explicit configuration)
- `create_issue` - Create new issue
- `comment_issue` - Add comment to issue
- `merge_pr` - Merge pull request
- `trigger_workflow` - Dispatch workflow

## Configuration

```toml
[github]
enabled = true
token = "ghp_..."  # or GITHUB_TOKEN env var
api_url = "https://api.github.com"  # for GitHub Enterprise
default_owner = "my-org"  # optional
allowed_actions = ["get_repo", "list_issues", "create_issue"]
```

## Status

**This feature is implemented** - See PR #XXXX

## Additional Context

- Related to: Composio integration (already supports GitHub via Composio)
- Similar pattern to: Jira integration, Notion integration
- Supports GitHub Enterprise by configuring api_url

### PR DESCRIPTION
**Pull request by vasanth53**
## Summary

- Base branch target: `master`
- Problem: ZeroClaw lacks native GitHub integration, requiring users to use browser automation or external tools
- Why it matters: GitHub is a critical platform for developers; native integration enables proper autonomous GitHub workflows
- What changed:
  - Added `GitHubConfig` to config schema with token, api_url, allowed_actions support
  - Created `GitHubTool` with 12 actions (get_repo, list_repos, list_issues, get_issue, create_issue, comment_issue, list_prs, get_pr, merge_pr, list_workflows, list_runs, trigger_workflow)
  - Updated integration registry from ComingSoon to Available/Active status
  - Added comprehensive unit and integration tests with wiremock
- What did not change: Other tools, channels, providers remain unchanged

## Label Snapshot

- Risk label: `risk: medium` (new external network calls)
- Size label: `size: M`
- Scope labels: `tool,integration`
- Module labels: `tool: github`

## Change Metadata

- Change type: `feature`
- Primary scope: `tool`

## Linked Issue

- Closes #4352

## Validation Evidence

```bash
cargo fmt --all -- --check  # Pass
cargo check                  # Pass
cargo test --lib github_tool # 12 tests passed
```

Test coverage:
- Unit tests: action allowlist, owner resolution, tool metadata
- Integration tests with wiremock: successful API calls, 404 errors, missing params, action not allowed

## Security Impact

- New permissions/capabilities? `Yes` - Tool can create/modify GitHub issues, PRs, and trigger workflows
- New external network calls? `Yes` - HTTPS calls to GitHub API
- Secrets/tokens handling changed? `Yes` - GitHub PAT stored/handled similarly to Jira API token
- If any `Yes`, describe risk and mitigation: Token stored encrypted at rest; actions controlled by allowed_actions config

## Privacy and Data Hygiene

- Data-hygiene status: `pass`
- Redaction/anonymization notes: All test data uses neutral placeholders (test-owner, test-repo, etc.)
- Neutral wording confirmation: `Yes`

## Compatibility / Migration

- Backward compatible? `Yes`
- Config/env changes? `No` - New [github] section optional
- Migration needed? `No`

## i18n Follow-Through

- i18n follow-through triggered? `No`

## Human Verification

- Verified scenarios:
  - All 12 unit/integration tests pass
  - Compilation with no warnings
  - Code formatting passes
- Edge cases checked:
  - Action not allowed error handling
  - Missing required params validation
  - HTTP error handling (404)
- What was not verified:
  - Live GitHub API calls (would require real credentials)

## Side Effects / Blast Radius

- Affected subsystems/workflows: New tool registration, new integration entry
- Potential unintended effects: None identified
- Guardrails/monitoring for early detection: Security policy integration for Act operations

## Agent Collaboration Notes

- Agent tools used: explore, read, edit, bash
- Workflow/plan summary: Created config schema, implemented tool following Jira/Notion patterns, added tests
- Verification focus: Testing with wiremock for isolated integration tests
- Confirmation: `Yes` - Followed AGENTS.md patterns for tool implementation

## Rollback Plan

- Fast rollback command/path: `git revert <commit>` or disable with `enabled = false`
- Feature flags or config toggles: `enabled` flag in [github] config
- Observable failure symptoms: Tool not registered if not enabled

## Risks and Mitigations

- Risk: External API dependency on GitHub could fail
  - Mitigation: Standard error handling with informative messages
- Risk: Token exposure if misconfigured
  - Mitigation: Standard encrypted storage pattern, masked in logs

### CODE REVIEW

File: src/tools/mod.rs [hunk 1]
```diff
@@ -45,6 +45,7 @@ pub mod file_read;
 pub mod file_write;
 pub mod gemini_cli;
 pub mod git_operations;
+pub mod github_tool;
```
Comment (ilteoood) [inline src/tools/mod.rs:48]: instead of `github_tool` I think it should be just `github`.
As `google_workspace`, for example is not `google_workspace_tool`

File: src/tools/github_tool.rs [hunk 2]
```diff
@@ -0,0 +1,1640 @@
...
+        self.allowed_actions.iter().any(|a| a == action)
+    }
+
+    fn resolve_owner(&self, owner: Option<&str>) -> Option<String> {
+        owner
+            .map(String::from)
+            .or_else(|| self.default_owner.clone())
+    }
+
+    async fn get_repo(&self, owner: &str, repo: &str) -> anyhow::Result<ToolResult> {
```
Comment (ilteoood) [inline src/tools/github_tool.rs:50]: instead of re-implementing the client from scratch, I think it would be better to use https://github.com/XAMPPRocky/octocrab
